### DISCUSSION TITLE
How to remove a file from stage area (`rm --cached`)

### DISCUSSION THREAD

**Discussion by noidilin** [jesseduffield/lazygit#3677] [Q&A] [answered]
I believe the equivalent command is `git rm --cached`,. I was expecting to find the similar command in the file panel, yet didn't find it. Am I missing anything?

**Comment by stefanhaller** [comment, marked as answer]
I'm not aware that this is possible with lazygit. What's the use case?

**Reply by noidilin** [reply]
This may be not a very useful use case, since I'm just start learning git.

Initially it was because I found that adding files into the `.gitignore` did not affect files that were already being tracked by git. Then I found `git rm --cached` can completely remove files from the stage area. When I tried to practice in lazygit, I realized that it didn't seem to have this command. 

This is probably a mistake that only newbies will encounter. Sorry to disturbing you.

**Reply by stefanhaller** [reply]
No, it's a valid request. It's just a situation that I personally haven't encountered yet, that's why I asked about the use case.

I'm not sure it's worth adding this to lazygit, given that it's probably somewhat rare to need it. Also, it's unclear how, since you usually don't see the file that you want to act on in lazygit (unless it's modified).

Personally, what I would have done in this situation is to rename the file, this will make it show up with a "D" in lazygit (and the renamed file with a question mark), then stage and commit the deleted file, then rename the file back to the original name.

**Reply by KennethHoff** [reply]
I've only ever used it in context of a .gitignore, so it would be nice if the ignore feature had a "also remove all tracked files that would've been ignored had they not already been tracked" option.

**Comment by arvid-berndtsson** [comment]
I would love to see this too
